histomicstk/segmentation/label/remove_overlap_nuclei.py

```diff
@@ -0,0 +1,58 @@
+import os
+from concurrent.futures import ProcessPoolExecutor
+
+import shapely
+from shapely.geometry import Polygon
+
+
+def create_polygon(coordinates):
+    """
+    Create a shapely Polygon from a list of points.
+
+    Args:
+        points (list): A list containing tuples representing the points of the polygon.
+
+    Returns:
+        shapely.geometry.Polygon: The polygon created from the points.
+    """
+    return Polygon(coordinates).buffer(0)
+
+
+def remove_overlap_nuclei(nuclei_list):
+    """
+    Remove overlapping nuclei from the given list using parallel processing.
+
+    This function creates a single shapely STRtree from all the polygons in the nuclei_list
+    and filters out the overlapping polygons.
+
+    Args:
+        nuclei_list (list): A list of dictionaries, each containing 'points' representing a polygon.
+
+    Returns:
+        list: A new list with overlapping nuclei removed.
+    """
+    # Extract all points from the nuclei_list to create polygons
+    coordinates_collection = [nuclei['points'] for nuclei in nuclei_list]
+
+    # Use ProcessPoolExecutor for parallel processing of polygon creation
+    with ProcessPoolExecutor(os.cpu_count()) as executor:
+        polygons = list(
+            executor.map(
+                create_polygon,
+                (coordinates for coordinates in coordinates_collection)))
```

Comment:
Is there a virtue of creating the `coordinates_collection`?  Couldn't you just do `nuclei['points'] for nuclei in nuclei_list`?  If not, then you should be able to skip this comprehension (e.g., `executor.map(create_polygon, (coordinates for coordinates in coordinates_collection))` should be the same as `executor.map(create_polygon, coordinates_collection)`)

Reply:
I agree, the `coordinate_collection` is redundant. I've replaced it with the `nuclei['points'] for nuclei in nuclei_list`